src/utils/guildRole.ts

```diff
@@ -135,3 +142,70 @@ export async function getGuildRoleByName(
   const roles = await getGuildRoles(env);
   return roles?.find((role) => role.name === roleName);
 }
+
+export async function mentionEachUserInMessage({
+  message,
+  userIds,
+  channelId,
+  env,
+}: {
+  message?: string;
+  userIds: string[];
+  channelId: number;
+  env: env;
+}) {
+  const batchSize = 5;
+  let waitTillNextAPICall = 0;
+  try {
+    const failedUsers: Array<string> = [];
+    for (let i = 0; i < userIds.length; i += batchSize) {
+      const batchwiseUserIds = userIds.slice(i, i + batchSize);
+      const messageRequest = batchwiseUserIds.map((userId) => {
+        return fetch(`${DISCORD_BASE_URL}/channels/${channelId}/messages`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bot ${env.DISCORD_TOKEN}`,
+          },
+          body: JSON.stringify({
+            content: `${message ? message + " " : ""} ${userId}`,
+          }),
+        }).then((response) => {
+          const rateLimitRemaining = parseRateLimitRemaining(response);
+          if (rateLimitRemaining === 0) {
+            waitTillNextAPICall = Math.max(
+              parseResetAfter(response),
+              waitTillNextAPICall
+            );
+          }
+          return response.json();
+        }) as Promise<discordMessageRequest | discordMessageError>;
+      });
+      const responses = await Promise.all(messageRequest);
+      responses.forEach((response, i) => {
+        if (response && "message" in response) {
+          failedUsers.push(batchwiseUserIds[i]);
+          console.error(`Failed to mention a user`);
+        }
+      });
+      await new Promise((resolve) =>
+        setTimeout(resolve, waitTillNextAPICall * 1000)
+      );
+      waitTillNextAPICall = 0;
+    }
+    if (failedUsers.length > 0) {
+      await fetch(`${DISCORD_BASE_URL}/channels/${channelId}/messages`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bot ${env.DISCORD_TOKEN}`,
+        },
+        body: JSON.stringify({
+          content: `Failed to tag ${failedUsers} individually.`,
+        }),
+      });
+    }
```

Comment:
What if the runtime stops before this?

Reply:
As the maximum execution time for waitUntil is 30 seconds, it is highly unlikely that something like this will happen.

Reply:
please add a comment or doc there mentioning the limitations and cases

Reply:
https://community.cloudflare.com/t/waituntil-settimeout-how-long-will-waituntil-wait-for/368393

I've attached the same link in the comment above.

<img width="853" alt="Screenshot 2024-03-16 at 8 03 29 AM" src="https://github.com/Real-Dev-Squad/discord-slash-commands/assets/56365512/216404ab-4e44-4508-8434-42906fa860a1">